catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java

```diff
@@ -873,4 +882,21 @@ public String getSortKey(String field) {
   public Stream<String> anyTextFields() {
     return anyTextFieldsCache.stream();
   }
+
+  /**
+   * Gets the metadata size limit from custom.system.properties. Defaults to 5mb limit.
+   *
+   * @return Integer metadata size limit in bytes
+   */
+  private Integer getMetadataSizeLimit() {
+    String sizeLimit = System.getProperty("metadata.size.limit");
+    if (StringUtils.isNumeric(sizeLimit)) {
+      if (Integer.parseInt(sizeLimit) * 1000000 < Integer.MAX_VALUE) {
+        METADATA_MAXIMUM_BYTES = Integer.parseInt(sizeLimit) * 1000000;
+        return METADATA_MAXIMUM_BYTES;
+      }
+    }
+    METADATA_MAXIMUM_BYTES = 5000000;
```

Comment:
✏️ Should make that value a constant and should it be 5242880 (i.e. 5 * 1024 * 1024)?